Automattic/jetpack
# [WIP] [DO NOT MERGE] Carousel: Use CSS Grid for fixed info footer, add icons

***Note: work on this change is progressing in #20145***

This is a WIP experiment, hacking around using CSS grid for laying out the info footer for the Jetpack Carousel. This exploration is branched from Ramon's work in https://github.com/Automattic/jetpack/pull/20108.

Possible fix for: Automattic/view-design#281 and resolves Automattic/view-design#284

#### Changes proposed in this Pull Request:

* Add an apparent fixed footer to the Jetpack Carousel, with info and comment buttons
* Hide the extra info area beneath the footer by default, to prevent vertical scrolling
* When the info or comment button is clicked, reveal

#### Screenshot of WIP

![carousel-footer-reveal-with-pages-sml](https://user-images.githubusercontent.com/14988353/122881065-1402cd00-d37e-11eb-9ce4-184448b227e4.gif)

#### To-dos

* [x] Should we hide the extra footer area when navigating back / forward through slides?
* [ ] Add smooth scrolling to scrolling up / down
* [ ] Add a timeout before toggling the show class _off_ so that we can smooth scroll up _before_ the extra info content is hidden
* [ ] Tweak the tap targets of the buttons so that they are larger — currently (with my thumbs at least) it's difficult to tap the buttons on mobile (iPhone 8)
* [ ] Tweak styling and positioning of buttons to better match designs
* [ ] Tweak styling, width and positioning of columns, info, captions, etc beneath the new footer
* [ ] Update styling to ensure it works in Light mode as well as the default Dark colour scheme
* [ ] Should we add in the description to the footer (as per the designs) or leave it off for the time being?
* [ ] Should we add in a number indicator in the bottom left (e.g. 1/5)?
* [ ] Invert the fill / background colours of the icon buttons when they're selected? (It looks like this is suggested from the designs)
* [ ] Conditionally hide the info and comment icons based on the site's Jetpack settings for showing Exif metadata / comments:

#### Jetpack product discussion


Designs: p9Jlb4-2cI-p2

#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Add an image gallery to a post on your test site
* With Jetpack active, and the Jetpack Carousel module activated, view the post on the front end of your site
* Click an image in the gallery to open up the carousel, and try out the info and comment buttons in the footer
* Go to the Jetpack settings and update the carousel to use the white colour scheme and check that the styling is correct (note: styling support for light theme hasn't been added yet) `/wp-admin/admin.php?page=jetpack#/writing`

![image](https://user-images.githubusercontent.com/14988353/122874729-a99a5e80-d376-11eb-811c-eadf6cf6750f.png)
